### DISCUSSION TITLE
RTMP Video Stream Install

### DISCUSSION THREAD

**Discussion by RadioHerzblut** [AzuraCast/AzuraCast#6807] [Q&A]
How do I set up an RTMP Stram in Azuracast ? 

Are there any instructions ?

**Comment by Moonbase59** [comment]
**Here be dragons.**

I have played around with video for some months, and it _can_ be done (Liquidsoap programming) but often an el-cheapo cloud VM can’t provide the computing power needed. Ideally, it should be done with hardware acceleration, which can be had at prohibitive cost only (or on home servers). On a 2–4 (dedicated) CPU cloud server, you _might_ get a 1280x720, 25 fps stream using ffmpeg/Liquidsoap with software encoding and much fine tuning, but you can easily ruin AzuraCast performance (stuttering, obscure errors, etc.) when using many overlays, fast-moving or hi-res/high framerate source video.

I think I have left some notes in the discussions here or at the Liquidsoap GitHub over time, but not yet published my code, because I’ll eventually switch to adding a hardware accel board into my server, just for that purpose. Personally, I’d use Intel VA-API/Quick Sync Video for that, because it works rather well on Linux. (I use it with _Jellyfin_ and _ErsatzTV_, as example.)

Nevertheless, BusterNeece has published some example code in a [Gist](https://gist.github.com/BusterNeece/5dbfb4dbc1846055c9ab07a7c685899c). Knowledge of Liquidsoap and video programming recommended.

- Some pointers from the [Discussions over at Liquidsoap](https://github.com/savonet/liquidsoap/discussions?discussions_q=is%3Aopen+video+involves%3AMoonbase59)

**Comment by Heimatradio** [comment]
I have Azuracast on a VPS on Ubuntu 22.04 with 6 CPU's and 16GB RAM.
I've been looking for step by step instructions for a long time to setup a video-stream on a icecast server with Azuracast.
Has anyone ever put this down on a digital paper?
I'm a little older now and can't do it alone anymore :-)
Thanks in advance

**Reply by sitedata** [reply]
Here's a guide for video streaming Icecast on AzuraCast that I found helpful: 
https://blog.radwebhosting.com/how-to-setup-video-streaming-with-icecast-on-azuracast-vps/

**Reply by gAlleb** [reply]
Ha! It's the most random unhelpful piece of text that I've read in months))
